pkg/driver/controller.go

```diff
@@ -278,24 +314,57 @@ func CreateLVMVolume(ctx context.Context, req *csi.CreateVolumeRequest,
 			}
 		}
 	}
+	var volObj *lvmapi.LVMVolume
+	var owner string             // owner node where volume is to be created
+	var lvmSnapshotCrName string // snapshot name if volume is to be created from snapshot
+	var volGroup string          // volume group where volume is to be created
+	// if volume is to be created from snapshot, verify request with snapshot values
+	if req.GetVolumeContentSource() != nil && req.GetVolumeContentSource().GetSnapshot() != nil {
+		snapId := req.GetVolumeContentSource().GetSnapshot().GetSnapshotId()
+		// snapshot ID is of the form <volume-name>@<snapshot-name>
+		parts := strings.Split(snapId, "@")
+		if len(parts) != 2 {
+			return nil, status.Errorf(codes.Internal,
+				"invalid snapshot ID %s", snapId)
+		}
+		lvmSnapCr, err := lvm.GetLVMSnapshot(parts[1])
+		if err != nil {
+			return nil, status.Errorf(codes.Internal, "snapshot %s not found: %s", parts[1], err.Error())
```

Comment:
Again as per csi, this should be NotFound error?

```suggestion
			return nil, status.Errorf(codes.NotFound, "snapshot %s not found: %s", parts[1], err.Error())
```